File: 백준/Bronze/23968. 알고리즘 수업 － 버블 정렬 1/알고리즘 수업 － 버블 정렬 1.py
import sys


def bubble_sort(lst, last, num):
    count = 0
    for n in range(last-1, 0, -1):
        for idx in range(n):
            if lst[idx] > lst[idx + 1]:
                count += 1
                lst[idx], lst[idx + 1] = lst[idx + 1], lst[idx]
                if count == num:
                    return lst[idx], lst[idx+1]
    return -1


N, k = map(int, sys.stdin.readline().split())
arr = list(map(int, sys.stdin.readline().strip().split()))
try:
    print(*bubble_sort(arr, N, k))
except:
    print(bubble_sort(arr, N, k))

File: node.js/백준/Bronze/23968. 알고리즘 수업 － 버블 정렬 1/알고리즘 수업 － 버블 정렬 1.js
const fs = require("fs");
const [[N, K], A] = fs.readFileSync("/dev/stdin")
                      .toString()
                      .trim()
                      .split("\n")
                      .map((line)=>line.split(" ").map(Number));

const bubbleSort = (arr) => {
    let k = 0;
    for (let last = N-1; last > 0; last --){
        for (let i = 0; i < last; i++){
            if (arr[i] > arr[i+1]) {
                k ++;
                if (k === K) return [arr[i+1], arr[i]].join(" ");
                [arr[i], arr[i+1]] = [arr[i+1], arr[i]];
            }
        }
    }
    return -1;
};

console.log(bubbleSort(A));